src/RouteLoader.js

```diff
@@ -18,16 +19,58 @@ type Props = {
   }
 }
 
-class RouteLoader extends Component<Props> {
+type State = {|
+  screenWidth: number,
+|}
+
+class RouteLoader extends Component<Props, State> {
+  constructor () {
+    super()
+    this.state = {
+      screenWidth: window.innerWidth,
+    }
+  }
+
+  componentWillMount () {
+    window.addEventListener('resize', this.handleWindowSizeChange)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('resize', this.handleWindowSizeChange)
+  }
+
+  handleWindowSizeChange = () => {
+    this.setState({ screenWidth: window.innerWidth })
+  }
+
+  checkIfMobile = () => {
+    if (/Android|webOS|iPhone|iPad|iPod|BlackBerry|BB|PlayBook|IEMobile|Windows Phone|Kindle|Silk|Opera Mini/i.test(navigator.userAgent)) {
```

Comment:
First link from google query "react check if device is mobile" – https://www.npmjs.com/package/react-device-detect
Please use this package instead. With **isMobile** from this package we don't even need to check screen width.
Moreover, if desktop user made his screen width less than 1024 we shouldn't show him DummyPage, because he is already using desktop.
Of course we should deal with layouts for the different screen widths, but that is other issue.